### DISCUSSION TITLE
Authelia + Haproxy - HOW TO

### DISCUSSION THREAD

**Discussion by rocklee44** [authelia/authelia#3398] [Q&A] [answered]
Hi all,
My environment : Centos 8 + haproxy-2.2.10 + lua-5.3.5 install from source following instruction : https://kifarunix.com/install-haproxy-on-rocky-linux-8/
I followed instruction at : https://www.authelia.com/docs/deployment/supported-proxies/haproxy.html
```
global
    # Path to haproxy-lua-http, below example assumes /usr/local/etc/haproxy/haproxy-lua-http/http.lua
    lua-prepend-path /usr/local/etc/haproxy/http.lua
    # Path to haproxy-auth-request
    lua-load /usr/local/etc/haproxy/auth-request.lua
    log stdout format raw local0 debug
```
When I check haproxy cfg I get errors:
```
haproxy -c -f haproxy.cfg
[NOTICE]   (26946) : haproxy version is 2.4.2-553dee3
[NOTICE]   (26946) : path to executable is /usr/local/sbin/haproxy
[ALERT]    (26946) : parsing [haproxy.cfg:5] : Lua runtime error: error loading module 'json' from file '/usr/local/etc/haproxy/http.lua':
        /usr/local/etc/haproxy/http.lua:673: too many C levels (limit is 200) in function at line 619 near 'v'

[ALERT]    (26946) : parsing [haproxy.cfg:51]: 'http-request' expects 'wait-for-handshake', 'use-service', 'send-spoe-group', 'sc-inc-gpc0(*)', 'sc-inc-gpc1(*)', 'sc-set-gpt0(*)', 'do-resolve(*)', 'cache-use', 'add-acl(*)', 'add-header', 'allow', 'auth', 'capture', 'del-acl(*)', 'del-header', 'del-map(*)', 'deny', 'disable-l7-retry', 'early-hint', 'normalize-uri', 'redirect', 'reject', 'replace-header', 'replace-path', 'replace-pathq', 'replace-uri', 'replace-value', 'return', 'set-header', 'set-log-level', 'set-map(*)', 'set-method', 'set-mark', 'set-nice', 'set-path', 'set-pathq', 'set-query', 'set-tos', 'set-uri', 'strict-mode', 'tarpit', 'track-sc(*)', 'set-timeout', 'wait-for-body', 'set-var(*)', 'unset-var(*)', 'set-priority-class', 'set-priority-offset', 'silent-drop', 'set-src', 'set-src-port', 'set-dst', 'set-dst-port', but got 'lua.auth-request'.
[ALERT]    (26946) : Error(s) found in configuration file : haproxy.cfg
```
How can I fix it ? Please give me some advice , thank you very much.

**Comment by james-d-elliott** [comment]
What's on line 5 of `haproxy.cfg`? I expect this is loading http.lua and the issue is something to do with the guide being wrong in some way.

**Comment by rocklee44** [comment]
Hi James,
line 5 of haproxy.cfg is : lua-load /usr/local/etc/haproxy/auth-request.lua (I edited my first post at haproxy cfg part - make it easier to view)
These are the top lines of the /usr/local/etc/haproxy/http.lua (I get this file from https://github.com/haproxytech/haproxy-lua-http as document)
```
local _author = "Adis Nezirovic <anezirovic@haproxy.com>"
local _copyright = "Copyright 2017-2020. HAProxy Technologies, LLC."
local _version = "1.0.0"

local json = require "json"
```

**Reply by james-d-elliott** [reply]
What `lua-json` version are you using? since that's the dep that's having the issue. Considering lua-http (the official HAProxy lua module) has not changed effectively in 12 months it's very unlikely to be this that's an issue, it's more likely the json package being used is significantly out of date since it was probably installed via CentOS.

**Reply by rocklee44** [reply]
I installed lua-5.3.5 from source and I'm not sure if lua-json is included.
My server Centos 8(minimal) has already lua-5.3.4 installed , and I also installed lua-json
```
rpm -qa | grep lua
lua-json-1.3.2-9.el8.noarch
lua-libs-5.3.4-12.el8.x86_64
lua-lpeg-1.0.1-6.el8.x86_64
lua-5.3.4-12.el8.x86_64
```

**Comment by rocklee44** [comment]


**Reply by james-d-elliott** [reply]
Are you saying it's fixed?

**Reply by rocklee44** [reply]
Sorry, my mistake , the issue is not fixed , I commented out `local json = require "json"` in /usr/local/etc/haproxy/http.lua to experiment.

**Reply by james-d-elliott** [reply]
So it's evidently a problem with that dep which haproxy's http.lua depends on if that removed the error, I've been trying to find an error with luajson but can't seem to find any similar. All so very strange.

**Comment by rocklee44** [comment, marked as answer]
I fixed it by editing /usr/local/etc/haproxy/http.lua
```
--local json = require "json"
local json = { _version = "1.3.2" }
```
It works fine now.
